src/Filter/AbstractDateFilter.php

```diff
@@ -68,16 +68,21 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $value
             if (false === $this->time) {
                 if ($value['value']['start'] instanceof \DateTime) {
                     $value['value']['start']->setTime(0, 0, 0);
+                } elseif ($value['value']['start'] instanceof \DateTimeImmutable) {
+                    $value['value']['start'] = $value['value']['start']->setTime(0, 0, 0);
```

Comment:
Maybe we could use that logic for `\DateTime` too, it would work I think.

Reply:
It would work, but since [`\DateTimeInterface`](https://www.php.net/manual/en/class.datetimeinterface.php) does not have `setTime` method, should we use it?

Reply:
Currently, there is no possibility for other classes to implement `\DateTimeInterface`. Do you propose to check explicitly against `\DateTime` and `\DateTimeimmutable`?

Reply:
Yes, that's what I'm proposing.

Reply:
Fine. I'll replace the check against `\DateTimeInterface` with 2 individual tests against `\DateTime` and `\DateTimeImmutable`.

Reply:
It's done.